core/sql/exp/exp_datetime.cpp

```diff
@@ -92,6 +92,30 @@ copyDatetimeFields(rec_datetime_field startField,
                    Lng32 dstLen,
                    NABoolean *roundedDownFlag);
 
+// Helper function to format extra error message text and report error
+// srcData is not null terminated, so need a buffer copy to build a C-style string
+// 
+static void 
+raiseDateConvErrorWithSrcData(int srcLen, ComDiagsArea** diagsArea, char *srcData, CollHeap *heap)
+{
+    char errstr[MAX_OFFENDING_SOURCE_DATA_DISPLAY_LEN];
+    memset(errstr, 0 , sizeof(errstr) );
+    if(srcLen > MAX_OFFENDING_SOURCE_DATA_DISPLAY_LEN -1 )
+      srcLen = MAX_OFFENDING_SOURCE_DATA_DISPLAY_LEN -1;
+    strncpy(errstr, srcData, srcLen);
+    ExRaiseSqlError(heap, diagsArea, EXE_CONVERT_DATETIME_ERROR,NULL,NULL,NULL,NULL,errstr);
+}
+
+// Helper function to format extra error message text and report error
+// 
+static void 
+raiseDateConvErrorWithSrcDataNumeric(ComDiagsArea** diagsArea, long srcData, CollHeap *heap)
+{
+   char errstr[MAX_OFFENDING_SOURCE_DATA_DISPLAY_LEN];
+   memset(errstr, 0 , sizeof(errstr) );
+   sprintf(errstr,"%ld",srcData);
```

Comment:
Minor comment: I try not to use sprintf anymore and use snprintf instead. Some static analysis tools flag this type of code as a potential buffer overrun, even though the length may be long enough.